### PR TITLE
soapdenovo2: strip optimization flags from injected flags

### PR DESCRIPTION
**Pull request by snehring**
This fixes an issue brought about by the recent change to compiler flag prioritization.

### CODE REVIEW

File: var/spack/repos/builtin/packages/soapdenovo2/package.py [hunk 1]
```diff
@@ -15,16 +17,19 @@ class Soapdenovo2(MakefilePackage):
...
 
     def flag_handler(self, name, flags):
-        if self.spec.satisfies("%gcc@10:"):
-            if name == "cflags" or name == "CFLAGS":
-                flags.append("-fcommon")
-            if name == "cxxflags" or name == "CXXFLAGS":
-                flags.append("-fcommon")
+        if name.lower() == "cflags" and self.spec.satisfies("%gcc@10:"):
+            flags.append("-fcommon")
+        if name.lower() == "cflags" or name.lower() == "cxxflags" or name.lower() == "cppflags":
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/soapdenovo2/package.py]: Could simplify this, e.g.,:

```suggestion
        if name.lower() in ["cflags", "cxxflags", "cppflags"]:
```

File: var/spack/repos/builtin/packages/soapdenovo2/package.py [hunk 2]
```diff
@@ -15,16 +17,19 @@ class Soapdenovo2(MakefilePackage):
...
+            opt_flag = re.compile("-O.*")
+            # This package cannot compile with anything other than the -O3
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/soapdenovo2/package.py]: Don't you mean the package cannot compile with any *optimization* flags/levels other than the `-O3` in the makefile?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
A suggestion and comment-related question.

File: var/spack/repos/builtin/packages/soapdenovo2/package.py (hunk 1 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/soapdenovo2/package.py]: Ha, yeah that would be cleaner.

File: var/spack/repos/builtin/packages/soapdenovo2/package.py (hunk 2 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/soapdenovo2/package.py]: Yep. I've changed the wording a bit to be more clear.

**Comment by snehring** [comment]
@tldahlgren is there anything else you'd like changed?

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**